paimon-core/src/main/java/org/apache/paimon/operation/FileDeletionBase.java

```diff
@@ -70,6 +73,13 @@ public abstract class FileDeletionBase {
 
     protected final Map<BinaryRow, Set<Integer>> deletionBuckets;
     protected final Executor ioExecutor;
+    protected boolean changelogDecoupled;
+
+    /** Used to record which tag is cached in tagged snapshots list. */
+    private int cachedTagIndex = -1;
+
+    /** Used to cache data files used by current tag. */
+    private final Map<BinaryRow, Map<Integer, Set<String>>> cachedTagDataFiles = new HashMap<>();
```

Comment:
maybe we should introduce a `ExpireDeletion`? `TagDeletion` will not use this field.

Reply:
It only be cached when accessed, So in `TagDeletion` it will not be populated

Reply:
My bad. TagDeletion will use skipper too.